CONTRIBUTING.md

```diff
@@ -137,9 +144,61 @@ The [packages/](packages/) directory contains the following public/published pac
+
+#### Integration Level Testing
+
+Integration testing ensures that Caliper integrates correctly with various packages, effectively testing the functionality of the package itself. These tests are organized within the caliper-tests-integration folder, with each test suite dedicated to a specific package or module.
+
+#### Testing Dependencies and Their Use
+
+The following are the recommended testing modules for this project. While these tools are commonly used in the current codebase, other appropriate tools may be used as long as they facilitate effective testing.
+
+* [mockery](https://www.npmjs.com/package/mockery): Mockery is a simple module for mocking Node.js modules during testing. It allows you to replace real modules with mocks or stubs.
+  
+* [mocha](https://mochajs.org/): Mocha is a feature-rich JavaScript test framework that runs on Node.js and in the browser. It facilitates asynchronous testing, making it easy to write simple and flexible tests.
+
+* [chai](https://www.npmjs.com/package/chai): Chai is a BDD/TDD assertion library for Node.js and the browser. It can be paired with any JavaScript testing framework. We use it to create readable and expressive assertions.
+
+* [sinon](https://sinonjs.org/releases/v18/): Sinon is a standalone test spies, stubs, and mocks for JavaScript. It works with any test framework and integrates well with Mocha and Chai. We utilize Sinon for checking how functions are called during testing.
+
+* [sinon-chai](https://www.npmjs.com/package/sinon-chai): This library provides a set of custom assertions for using Sinon with Chai. It allows you to write more readable assertions for Sinon spies, stubs, and mocks.
+
+* [nyc](https://www.npmjs.com/package/nyc): NYC is a command-line utility for generating code coverage reports. It is often used with Mocha to ensure that tests cover as much code as possible.
+
+*[yeoman-generator](https://www.npmjs.com/package/yeoman-generator): Rails-inspired generator system that provides scaffolding for your apps
+
+#### Mandatory Tools in the Testing Pipeline
+
+In addition to the testing frameworks, the following tools are mandatory for all testing pipelines:
+
+* [eslint](https://eslint.org/): ESLint is a static code analysis tool for identifying problematic patterns in JavaScript code. It is essential for maintaining code quality.
+* [license-check-and-add](https://www.npmjs.com/package/license-check-and-add): This tool ensures that all files in the codebase contain the required license headers. It is mandatory for all code submissions.
+
+#### Points to Note for Adding a Conforming Test
```

Comment:
This list really only applies to unit testing at the moment. I don't see they would be relevant for integration testing so we need to make that clear or restructure to subcategorise this under unit testing